### PR TITLE
769 Resolved Unit Test Guidance and Programming Strategy overlap

### PR DESCRIPTION
**Pull request by millerg23**
Moved info from Unit Testing section in Programming Strategy Vignette into Unit Test Guidance Vignette, and replaced with link to Unit Test Guidance.

### CODE REVIEW

File: vignettes/programming_strategy.Rmd [hunk 1]
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
...
-* Input and output should be as simple as possible.
-* Values should be hard-coded whenever possible.
-* If values need to be derived, only unit tested functions can be used.
-
-If a dataset needs to be created for testing purpose, it should be done so using the function `tibble::tribble()`. 
-Make sure to align columns as well. This ensures quick code readability. 
-
-If a unit test fails, it must provide a helpful error message, e.g., if the output dataset is not as expected, the observations, variables, and values which differ should be provided. See also test helper function (link to be added).
+A function requires a set of unit tests to verify it produces the expected result.
+See [Writing Unit Tests in {admiral}](unit_test_guidance.html#Writing Unit Tests in {admiral}) for details.
```
Comment (bundfussr) [inline vignettes/programming_strategy.Rmd:378]: The link does not work. `unit_test_guidance.html#writing-unit-tests-in-admiral-` should work.

File: vignettes/programming_strategy.Rmd [hunk 2]
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
...
 # Unit Testing
 
-A function requires a set of unit tests to verify it produces the expected result. 
-If a function is defined in a file named `R/derive_baseline.R` then its accompanying unit tests should be stored in `tests/testthat/test-derive_baseline.R`.
-
-The testing framework to be used is [testthat](https://testthat.r-lib.org/).
-
-Unit tests should cover the functionality of the function.
-If another function g is called within a function f, the unit tests of f should not test the functionality of g. 
-This should be tested by the unit tests of g, i.e. unit tests should be added at the lowest level.
```
Comment (bundfussr) [inline vignettes/programming_strategy.Rmd:384]: I think we lost this part. Was this intended?

File: vignettes/programming_strategy.Rmd [hunk 3]
```diff
@@ -374,25 +374,8 @@ where the labels are stored.
...
-The input and expected output for the unit tests must follow the following rules:
-
-* Input and output should be as simple as possible.
-* Values should be hard-coded whenever possible.
-* If values need to be derived, only unit tested functions can be used.
-
-If a dataset needs to be created for testing purpose, it should be done so using the function `tibble::tribble()`. 
-Make sure to align columns as well. This ensures quick code readability. 
-
-If a unit test fails, it must provide a helpful error message, e.g., if the output dataset is not as expected, the observations, variables, and values which differ should be provided. See also test helper function (link to be added).
```
Comment (bundfussr) [inline vignettes/programming_strategy.Rmd:395]: I think this point got lost. Was this intended?

I would replace "See also test helper function (link to be added)." with "See `expect_dfs_equal()` for comparing datasets."

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: vignettes/programming_strategy.Rmd (hunk 2 above)
Comment (rossfarrugia) [reply vignettes/programming_strategy.Rmd:384]: agree - i would keep as things like the naming convention are important reminders here

**Review by rossfarrugia — CHANGES_REQUESTED** [review]
Agree with Stefan's requests - nothing else to add

**Comment by millerg23** [comment]
@bundfussr I fixed the typo in link to Unit Test Guidance in the Programming Strategy Vignette.
Also, added a section 'Test Coverage' in the Unit Test Guidance.
Do you think any further updates required, regarding `expect_dfs_equal()`

Thanks!

**Review by bundfussr — APPROVED** [review]

**Review by thomas-neitmann — APPROVED** [review]

**Pull request merged**